account_invoice_comment_template/models/account_move.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2014 Guewen Baconnier (Camptocamp SA)
+# Copyright 2013-2014 Nicolas Bessi (Camptocamp SA)
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2019 Iván Todorovich (Druidoo)
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    comment_template1_id = fields.Many2one(
+        "base.comment.template",
+        string="Top Comment Template",
+        domain=[
+            ("model_ids.model", "=", "account.move"),
+            ("position", "=", "before_lines"),
+        ],
+    )
+
+    comment_template2_id = fields.Many2one(
+        "base.comment.template",
+        string="Bottom Comment Template",
+        domain=[
+            ("model_ids.model", "=", "account.move"),
+            ("position", "=", "after_lines"),
+        ],
+    )
+
+    note1 = fields.Html("Top Comment")
+    note2 = fields.Html("Bottom Comment")
+
+    @api.onchange("comment_template1_id")
+    def _set_note1(self):
+        comment = self.comment_template1_id
+        if comment:
+            self.note1 = comment.text
+
+    @api.onchange("comment_template2_id")
+    def _set_note2(self):
+        comment = self.comment_template2_id
+        if comment:
+            self.note2 = comment.text
+
+    @api.depends("partner_id")
+    def get_comment_template_item(self, position):
+        return self.env["base.comment.template"].search(
+            [
+                ("model_ids.model", "=", "account.move"),
+                ("position", "=", position),
+                ("partner_ids", "in", self.partner_id.ids),
+            ],
+            limit=1,
+        )
```

Comment:
Thanks, i have changed to add `company_id` in domain and order by priority.
It's not possible to use `get_comment_template` function because need `base.comment.template` record and these function not return it. Maybe i have to create a little PR about it.

Reply:
With these change https://github.com/OCA/reporting-engine/pull/497/ it's possible to change it something like these:

```
_name = "account.move"
_inherit = ["account.move", "comment.template"]

self.get_comment_template_records(
    position="before_lines", company_id=self.company_id.id, partner_id=self.partner_id.id
)[0]
```

Reply:
I think that would be good. I have already approved that PR.